data/campaigns/fri05.wmf/scripting/mission_thread.lua

```diff
@@ -0,0 +1,235 @@
+               end
+            end
+         end
+      end
+      sleep(7200)
+   end
+end
+
+function check_objective_wood()
+   local o = add_campaign_objective(obj_wood)
+   while not check_for_buildings(p1, {frisians_woodcutters_house = 3, frisians_foresters_house = 4}) do
+      sleep(4273)
+   end
+   set_objective_done(o)
+   campaign_message_box(secured_wood)
+end
+
+function check_objective_block()
+   local o = add_campaign_objective(obj_block)
+   while true do
+      sleep(4916)
+      local sites = 0
+      for x = 84, 94 do
+         for y = 114, 122 do
+            local f = map:get_field(x, y)
+            if f.immovable ~= nil and f.immovable.descr.type_name == "militarysite" and f.immovable.fields[1] == f and f.owner == p1 then
+               sites = sites + 1
+               if sites >= 3 then
+                  set_objective_done(o)
+                  campaign_message_box(secured_block)
+                  return
+               end
+            end
+         end
+      end
+   end
+end
+
+function check_objective_uplands()
+   local o = add_campaign_objective(obj_uplands)
+   while true do
+      sleep(4471)
+      local nr = 0
+      for _,descr in pairs(p1.tribe.buildings) do
+         for _,b in pairs(p1:get_buildings(descr.name)) do
+            if b.fields[1].height > 40 then
+               nr = nr + 1
+               if nr >= 15 then
+                  set_objective_done(o)
+                  campaign_message_box(secured_uplands)
+                  return
+               end
+            end
+         end
+      end
+   end
+end
+
+function mission_thread()
+   include "map:scripting/init_p2.lua"
+   run(check_fish)
+   campaign_message_box(intro_1)
+   campaign_message_box(intro_2)
+   campaign_message_box(intro_3)
+   campaign_message_box(intro_4)
+
+   scroll_to_field(map.player_slots[1].starting_field)
+   include "map:scripting/init_p1.lua"
+   p1:reveal_fields(map.player_slots[2].starting_field:region(9))
+   local reveal_fields = {}
+   for x=0,121 do
+      for y=17,240 do
+         local f = map:get_field(x, y)
+         p2:reveal_fields({f})
+         if p1:sees_field(f) then
+            p1:reveal_fields({f})  -- Make currently seen fields permanently visible without hiding them first
+         else
+            table.insert(reveal_fields, f)
+         end
+      end
+   end
+   run(function()
+      reveal_randomly(p1, reveal_fields, 100)
+      reveal_fields = nil
+   end)
+   sleep(5000)
+
+   campaign_message_box(intro_5)
+   campaign_message_box(intro_6)
+   campaign_message_box(intro_7)
+
+   scroll_to_field(map.player_slots[2].starting_field)
+   sleep(5000)
+   scroll_to_field(map.player_slots[1].starting_field)
+
+   campaign_message_box(intro_8)
+   campaign_message_box(intro_9)
+   campaign_message_box(intro_10)
+   local o = add_campaign_objective(obj_fight)
+   sleep(1500)
+
+   campaign_message_box(trade_1)
+   campaign_message_box(trade_2)
+
+   for i,f in ipairs({
```

Comment:
what is the purpose of this camera move? it is not clear to the player what shall be shown.

Reply:
The idea was to give the player a brief tour of the island, so when afterwards the campaign characters speak about the Uplands and Lowlands and where the land is good the player will already have gotten some impressions so he may know what they mean. But it's just a few glances; should I remove it?

Reply:
no I like the feature, and now that I am aware of its purpose I can and will make some sggestions.